### ISSUE TITLE
NewPipe crashes when de/activating subtitles in Samsung Dex mode

### ISSUE THREAD

**Issue by hammolo** [TeamNewPipe/NewPipe#7391]
### Checklist

- [x] I am using the latest version 
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug
open Newpipe in Samsung Dex (I tried it Dex for Windows, probably also happens in native Dex), watch a video, try to activate or deactivate subtitles; crash ensues.

### Device info

 - Android version/Custom ROM version: Android 11
 - Device model: Samsung Galaxy S21+

**Comment by litetex** [comment]
I don't know if NewPipe is built for Samsung Dex in the first place as it's some kind of "custom manufacturer Desktop-Android extension" (https://en.wikipedia.org/wiki/Samsung_DeX).

Anyway we can't fix anything with the poor description of the issue.

Are there any logs/screen recordings etc. available?
Also what is meant with "Dex, subtitles, crash"?

**Comment by hammolo** [comment]
Sorry, it was my first bug report ever.
I don't think anyone tested specifically NewPipe under Samsung Dex, but apparently it works quite well except for this problematic bug. This happens when clicking on the activate/deactivate subtitles button, or when trying to change subtitles language.
Here is the log:

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ IT
* __Content Language:__ it-IT
* __App Language:__ it_IT
* __Service:__ none
* __Version:__ 0.21.13
* __OS:__ Linux samsung/t2sxeea/t2s:11/RP1A.200720.012/G996BXXU3AUIE:user/release-keys 11 - 30
<details><summary><b>Crash log </b></summary><p>

```
android.view.WindowManager$BadTokenException: Unable to add window -- token android.view.ViewRootImpl$W@fda8ca is not valid; is your activity running?
	at android.view.ViewRootImpl.setView(ViewRootImpl.java:1444)
	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:469)
	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:114)
	at android.widget.PopupWindow.invokePopup(PopupWindow.java:1657)
	at android.widget.PopupWindow.showAsDropDown(PopupWindow.java:1460)
	at android.widget.ListPopupWindow.show(ListPopupWindow.java:771)
	at com.android.internal.view.menu.CascadingMenuPopup.showMenu(CascadingMenuPopup.java:480)
	at com.android.internal.view.menu.CascadingMenuPopup.show(CascadingMenuPopup.java:289)
	at com.android.internal.view.menu.MenuPopupHelper.showPopup(MenuPopupHelper.java:303)
	at com.android.internal.view.menu.MenuPopupHelper.tryShow(MenuPopupHelper.java:192)
	at com.android.internal.view.menu.MenuPopupHelper.show(MenuPopupHelper.java:156)
	at android.widget.PopupMenu.show(PopupMenu.java:230)
	at org.schabi.newpipe.player.Player.onCaptionClicked(Player.java:3563)
	at org.schabi.newpipe.player.Player.onClick(Player.java:3670)
	at android.view.View.performClick(View.java:8160)
	at android.widget.TextView.performClick(TextView.java:16222)
	at android.view.View.performClickInternal(View.java:8137)
	at android.view.View.access$3700(View.java:888)
	at android.view.View$PerformClick.run(View.java:30236)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:246)
	at android.app.ActivityThread.main(ActivityThread.java:8633)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)

```
</details>
<hr>

**Comment by litetex** [comment]
@hammolo 

Thank you for the additional information 😄

The bug looks similar to #5264, however the issue was closed because the author didn't respond.

**Comment by litetex** [comment]
Issue was merged (with similar ones) into #7865.
Please go on with the conversation inside there.
